crates/aiken-project/src/lib.rs

```diff
@@ -702,14 +710,60 @@ where
             cpu: i64::MAX,
         };
 
-        scripts
-            .into_iter()
-            .filter(|script| -> bool {
-                let path = format!("{}{}", script.module, script.name);
+        let scripts = if let Some(match_tests) = match_tests {
+            let match_tests: Vec<(&str, Option<Vec<String>>)> = match_tests
+                .iter()
+                .map(|match_test| {
+                    let mut match_split_dot = match_test.split('.');
 
-                !matches!(&match_name, Some(search_str) if !path.contains(search_str))
-            })
-            .collect::<Vec<Script>>()
+                    let match_module = match_split_dot.next().unwrap_or("");
```

Comment:
```suggestion
                    let match_module = if match_test.contains('.') {
                        match_split_dot.next().unwrap_or("")
                    } else {
                        ""
                    };
```

So that `-m foo` behaves the same as `-m .{foo}`